R/zchunk_L244.building_det.R

```diff
@@ -0,0 +1,1040 @@
+      # Split by SSP, creating a list with a tibble for each SSP, then add attributes
+      split(.$SSP) %>%
+      lapply(function(df){
+        select(df, -SSP) %>%
+          add_units("Unitless") %>%
+          add_comments("Satiation adder compute using satiation level, per-capita GDP and per-capita floorsapce") %>%
+          add_precursors("energy/A44.satiation_flsp_SSPs", "energy/A44.gcam_consumer", "common/GCAM_region_names",
+                         "energy/A_regions", "L102.pcgdp_thous90USD_Scen_R_Y", "L101.Pop_thous_R_Yh",
+                         "L144.flsp_bm2_R_res_Yh", "L144.flsp_bm2_R_comm_Yh")
+      })
+
+    # Assign each tibble in list
+    for(i in names(L244.SatiationAdder_SSPs)) {
+      assign(paste0("L244.SatiationAdder_", i), L244.SatiationAdder_SSPs[[i]] %>%
+               add_title(paste0("Satiation adders in floorspace demand function: ", i)) %>%
+               add_legacy_name(paste0("L244.SatiationAdder_", i)))
+    }
+
+    # L244.GenericBaseService and L244.ThermalBaseService: Base year output of buildings services (per unit floorspace)
+
+    # First, separate the thermal from the generic services. Generic services will be assumed to produce
+    # internal gain energy, so anything in the internal gains assumptions table will be assumed generic
+    generic_services <- unique(A44.internal_gains$supplysector)
+    thermal_services <- setdiff(unique(A44.sector$supplysector), generic_services)
+
+    # Base-service: filter only the model base years and change names as indicated in calibrated_techs_bld_det
+    L244.base_service <- L144.base_service_EJ_serv %>%
+      rename(base.service = value) %>%
+      mutate(base.service = round(base.service, energy.DIGITS_CALOUTPUT)) %>%
+      filter(year %in% BASE_YEARS) %>%
+      left_join_keep_first_only(calibrated_techs_bld_det, by = c("sector", "service")) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      select(c("region", "gcam.consumer", "nodeInput", "building.node.input"),
+             building.service.input = supplysector, year, base.service) # replace with LEVEL2_DATA_NAMES[["BldNodes]])
+
+    # Separate thermal and generic services into separate tibbles
+    L244.GenericBaseService <- L244.base_service %>%
+      filter(building.service.input %in% generic_services)
+
+    L244.ThermalBaseService <- L244.base_service %>%
+      filter(building.service.input %in% thermal_services) %>%
+      rename(thermal.building.service.input = building.service.input)
+
+    # L244.HDDCDD: Heating and cooling degree days by scenario
+    L244.all_sres_gcm <- unique(L143.HDDCDD_scen_R_Y[c("SRES", "GCM")]) # These HDD/CDD scenarios are pretty old, should be updated eventually
+
+    L244.HDDCDD_scen_R_Y <- L143.HDDCDD_scen_R_Y %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID")
+
+    # Let's make a climate normal for each region, using a selected interval of years
```

Comment:
"climate normal" seems odd. Is this a no climate change file? If so, maybe say that.

Reply:
Climate normal = average climatic conditions in historical years. This is calculated from climate model output tho; at the time we didn't have any re-analysis data available, and when we did get that, it didn't connect at all with the climate model outputs. E.g. historical HDD of Colombia was about 1000 in this climate model, and was <10 in the re-analysis data. We never committed the re-analysis data to the core model.

Reply:
Can we call it "historical" or "average" or "contemporary"? "normal" seems abnormal to me.

Reply:
Uh, obvi this isn't super important but "climate normal" is the technical term. E.g., https://www.ncdc.noaa.gov/data-access/land-based-station-data/land-based-datasets/climate-normals

Reply:
Had no idea. I guess it is fine then.